internal/native_image/rules.bzl

```
@@ -167,7 +211,9 @@ def _graal_binary_implementation(ctx):
             collect_default = True,
             files = [],
         ),
-    )]
+    )] + (
+        cc_info and [cc_info] or []
```

I personally find this easier to read:
```suggestion
        [cc_info] if cc_info else []
```